mlflow/models/utils.py

```diff
@@ -262,97 +273,136 @@ def __init__(self, input_example: ModelInputExample):
+                    raise TypeError(
+                        "Expected one of the following types:\n"
+                        "- pandas.DataFrame\n"
+                        "- numpy.ndarray\n"
+                        "- dictionary of (name -> numpy.ndarray)\n"
+                        "- scipy.sparse.csr_matrix\n"
+                        "- scipy.sparse.csc_matrix\n"
+                        "- dict\n"
+                        "- list\n"
+                        "- scalars\n"
+                        f"but got '{type(input_example)}'",
+                    )
+                self.data = _convert_dataframe_to_split_dict(self._inference_data)
+                self.serving_input = {DF_SPLIT: self.data}
+                orient = "split" if "columns" in self.data else "values"
                 self.info.update(
                     {
                         "type": "dataframe",
-                        "pandas_orient": "split",
+                        "pandas_orient": orient,
                     }
                 )
-        else:
-            self._inference_data = _coerce_to_pandas_df(input_example)
-            if self._inference_data is None:
-                raise TypeError(
-                    "Expected one of the following types:\n"
-                    "- pandas.DataFrame\n"
-                    "- numpy.ndarray\n"
-                    "- dictionary of (name -> numpy.ndarray)\n"
-                    "- scipy.sparse.csr_matrix\n"
-                    "- scipy.sparse.csc_matrix\n"
-                    "- dict\n"
-                    "- list\n"
-                    "- scalars\n"
-                    f"but got '{type(input_example)}'",
-                )
-            self.data = _handle_dataframe_input(self._inference_data)
-            orient = "split" if "columns" in self.data else "values"
-            self.info.update(
-                {
-                    "type": "dataframe",
-                    "pandas_orient": orient,
-                }
-            )
 
-    def save(self, parent_dir_path: str):
-        """Save the example as json at ``parent_dir_path``/`self.info['artifact_path']`."""
         if self._inference_params is not None:
-            data = {EXAMPLE_DATA_KEY: self.data, EXAMPLE_PARAMS_KEY: self._inference_params}
+            self.data = {EXAMPLE_DATA_KEY: self.data, EXAMPLE_PARAMS_KEY: self._inference_params}
+            if is_unified_llm_input:
+                self.serving_input = {
+                    **(self.serving_input or {}),
+                    **self._inference_params,
+                }
+            else:
+                self.serving_input = {
+                    **(self.serving_input or {}),
+                    SERVING_PARAMS_KEY: self._inference_params,
+                }
+
+        self.json_data = json.dumps(self.data, cls=NumpyEncoder)
+        if self.serving_input:
+            self.json_serving_input = json.dumps(self.serving_input, cls=NumpyEncoder)
```

Comment:
```suggestion
            self.json_serving_input = json.dumps(self.serving_input, cls=NumpyEncoder, indent=2)
```

can we add `indent=2` for better readability?